modules/thermalsensor_generic.c

```diff
@@ -0,0 +1,1451 @@
+static char *
+tsg_util_slice_str(char **ppos)
+{
+  return tsg_util_slice_token(*ppos, ppos, -1);
+}
+
+/** Parse integer value from config line
+ *
+ * Slices string terminated by any whitespace character
+ * from parse position and convert it to integer.
+ *
+ * Parse position is updated so that it points to data
+ * remaining after sliced value.
+ *
+ * @param ppos parse position
+ *
+ * @return integer value
+ */
+static int
+tsg_util_slice_int(char **ppos)
+{
+  return strtol(tsg_util_slice_str(ppos), 0, 0);
+}
+
+/** Helper for setting dynamic string variable
+ *
+ * @param ptr  pointer to string variable
+ * @param val  string, or NULL
+ */
+static void
+tsg_util_set_string(char **ptr, const char *val)
+{
+    char *use = (val && *val) ? strdup(val) : 0;
+    free(*ptr), *ptr = use;
+}
+
+/** Get content of a text file as a string
+ *
+ * @param path  file path
+ *
+ * @return content of file, or NULL in case of errors
+ */
+static char *
+tsg_util_read_file(const char *path)
+{
+    char *text = 0;
+    int   file = -1;
+    char *buff = 0;
+    int   size = 512;
+    int   used = 0;
+
+    file = TEMP_FAILURE_RETRY(open(path,O_RDONLY));
+    if( file == -1 )
+        goto EXIT;
+
+    buff = malloc(size + 1);
+    if( !buff )
+        goto EXIT;
+
+    for( ;; ) {
+        int want = size - used;
+        int have = TEMP_FAILURE_RETRY(read(file, buff+used, want));
+
+        if( have < 0 )
+            goto EXIT;
+
+        used += have;
+
+        if( have < want )
+            break;
+
+        size = size * 2;
+
+        char *temp = realloc(buff, size + 1);
+        if( !temp )
+            break;
+
+        buff = temp;
+    }
+
+    buff[used] = 0;
+
+    text = buff, buff = 0;
+
+EXIT:
+    free(buff);
+
+    if( file != -1 )
+        TEMP_FAILURE_RETRY(close(file));
+
+    return text;
+}
+
+/** Write string to an already existing file
+ *
+ * @param path  file path
+ * @param text  content to write
+ *
+ * @return true if all data could be written, false otherwise
+ */
+static bool
+tsg_util_write_file(const char *path, const char *text)
+{
+    bool ack  = false;
+    int  file = -1;
+    int  todo = strlen(text);
+
+    file = TEMP_FAILURE_RETRY(open(path,O_WRONLY));
+    if( file == -1 )
+        goto EXIT;
+
+    while( todo > 0 ) {
+        int done = TEMP_FAILURE_RETRY(write(file, text, todo));
+        if( done < 0 )
+            goto EXIT;
+
+        text += done;
+        todo -= done;
+    }
+
+    ack = true;
+
+EXIT:
+
+    if( file != -1 )
+        TEMP_FAILURE_RETRY(close(file));
+
+    return ack;
+}
+
+/** Convert string to integer value
+ *
+ * The string must contain number and nothing else
+ * but a number.
+ *
+ * @param text    string to convert
+ * @param value   where to store integer value
+ *
+ * @return true if value was obtained, false otherwise
+ */
+static bool
+tsg_util_parse_int(const char *text, int *value)
+{
+    bool  ack = false;
+    char *end = 0;
+    int   val = strtol(text, &end, 0);
```

Comment:
Locales make using strtod() really problematic, but IIRC strtol() should be affected only via isspace(). Which should not be an issue because any leading white space is removed before we end up here.

Also: being a system daemon, dsme runs in the C locale.